components/public-api/typescript-common/src/public-api-converter.ts

```diff
@@ -1211,6 +1213,47 @@ export class PublicAPIConverter {
                 );
             }
         }
+
+        const capitalize = (input: string) => {
+            return input.charAt(0).toUpperCase() + input.slice(1);
+        }
+
+        // This is a hack mimicking the supervisor behavior of adding dynamic IDE tasks https://github.com/gitpod-io/gitpod/blob/e7d79c355e2cd6ac34056ea52d7bdcda45975839/components/ide-service/pkg/server/server.go#L508-L540
+        const jetbrainsIdes = prebuild.workspace.config.jetbrains;
+        if (jetbrainsIdes) {
+            for (const ide in jetbrainsIdes) {
+                const ideConfig = jetbrainsIdes[ide as keyof typeof jetbrainsIdes]!;
+                if (!ideConfig.prebuilds) {
+                    continue;
+                }
+
+                if (ideConfig.prebuilds.version === "latest" || ideConfig.prebuilds.version === "both") {
+                    tasks.push(
+                        new TaskLog({
+                            taskId: `jb-warmup-${ide}-latest`,
+                            taskLabel: `JetBrains ${capitalize(ide)} warmup (latest)`,
```

Comment:
Let's ignore ide names or sort them both in supervisor and this converter?

Reply:
@mustard-mh do we sort them anywhere else in the UI?

Reply:
> do we sort them anywhere else in the UI?

what do you mean?

warmup tasks is added in supervisor with the ordering of `readDir`'s output. If we don't sync the way they sort tasks, the output will mismatch. i.e. Goland's output is shown in IntelliJ 's tab

Reply:
@mustard-mh oh I see what you meant now by your original comment. I have not resorted to sorting here, because it seems we go through a stable order of IDEs on the go side from the configmap:
https://github.com/gitpod-io/gitpod/blob/0bc104701dd89fc0ee82160d05f14fb8bce090b2/install/installer/pkg/components/ide-service/ide-configmap.json#L612-L622

and on the TS types side:
https://github.com/gitpod-io/gitpod/blob/6dfe1775347e4e96be3f84cd48b9e377dfae5cd8/components/gitpod-protocol/src/protocol.ts#L789-L799

Both sides should go through de-facto the same list and therefore not require sorting.

Reply:
The ordering of supervisor is depends on [ReadDir](https://github.com/gitpod-io/gitpod/blob/e8f9067c5503d77a38197aea0c7e52418b4712b9/components/supervisor/pkg/supervisor/config.go#L575-L620)

PS. object has no ordering in nodejs

Reply:
I could reproduce the misordering here, thanks for pointing it out!

Reply:
I did simple lexographical sorting in https://github.com/gitpod-io/gitpod/pull/19945/commits/97a4ea0a4fc6935262cfb370bdd4a60bd415b571, but I'm open to suggestions if there is an easy way of getting some static order on both sides which we don't have to hardcode.